client/main.lua

```diff
@@ -210,30 +193,67 @@ CreateThread(function()
                         })
                     end
                 else
-                    scrapPoly[#scrapPoly + 1] = BoxZone:Create(vector3(v.coords.x, v.coords.y, v.coords.z), v.length, v.width, {
-                        heading = v.heading,
-                        name = k..i,
-                        debugPoly = false,
-                        minZ = v.coords.z - 1,
-                        maxZ = v.coords.z + 1,
-                    })
-                    local scrapCombo = ComboZone:Create(scrapPoly, {name = "scrapPoly"})
-                    scrapCombo:onPlayerInOut(function(isPointInside)
-                        local inVehicle = IsPedInAnyVehicle(cache.ped, false)
-                        if isPointInside then
-                            if not isBusy then
-                                if k == 'deliver' and inVehicle then
-                                    exports['qbx-core']:DrawText(Lang:t('text.disassemble_vehicle'),'left')
-                                else
-                                    exports['qbx-core']:DrawText(Lang:t('text.email_list'),'left')
-                                end
-                                keyListener(k)
+                    if k == 'deliver' then
+                        local function onEnter()
+                            if IsPedInAnyVehicle(cache.ped, false) and not isBusy then
+                                exports['qbx-core']:DrawText(Lang:t('text.disassemble_vehicle'),'left')
+                                CreateThread(function()
+                                    while true do
+                                        if IsControlPressed(0, 38) then
+                                            exports['qbx-core']:KeyPressed()
+                                            exports['qbx-core']:HideText()
```

Comment:
KeyPressed hides the text with a 500ms delay. You only need one or the other, not both. I recommend using HideText, since KeyPressed is an inappropriately named function given its contents. Should probably be deprecated.

Reply:
Corrected